mapie/tests/test_calibration.py

```diff
@@ -0,0 +1,393 @@
+from typing import Union
+import numpy as np
+import pandas as pd
+from sklearn.calibration import _SigmoidCalibration
+from sklearn.pipeline import make_pipeline, Pipeline
+from sklearn.compose import ColumnTransformer
+from sklearn.impute import SimpleImputer
+from sklearn.preprocessing import OneHotEncoder
+from sklearn.linear_model import LinearRegression, LogisticRegression
+from sklearn.base import RegressorMixin, ClassifierMixin
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.datasets import make_classification
+from sklearn.model_selection import train_test_split
+
+from mapie.calibration import MapieCalibrator
+from mapie.metrics import top_label_ece, expected_calibration_error
+from inspect import signature
+import pytest
+
+random_state = 20
+
+CALIBRATORS = [
+    "sigmoid", "isotonic", _SigmoidCalibration(), LinearRegression()
+]
+
+ESTIMATORS = [
+    LogisticRegression(),
+    RandomForestClassifier(random_state=random_state),
+]
+
+results = {
+    "y_score": [
+        [0, 0.66666667, 0],
+        [0.66666667, 0, 0],
+        [0, 0.66666667, 0],
+        [0, 0.66666667, 0],
+        [0, 0.66666667, 0],
+        [0, 0, 0.35635314],
+        [0, 0, 0.18501723],
+    ],
+    "top_label_ece": 0.277,
+}
+
+results_binary = {
+    "y_score": [
+        [0.83169648, 0],
+        [0.3333333, 0],
+        [0, 0.66666667],
+        [0.82329671, 0],
+        [0, 0.66666667],
+        [0.88521744, 0],
+        [0.84504048, 0],
+        [0.6666663, 0],
+        [0, 0.66666667],
+        [0, 0.66666667],
+    ],
+    "top_label_ece": 0.273451,
+    "ece": 0.471858484018367,
+}
+
```

Comment:
be careful with those examples, did you compute this by hand ? This is how you should test your algo, the goal is not only to check is the results change over time but to check you have the expected results. If those results come from your code, please compute by hand to check

Reply:
Now I have. I can confirm the results are the same.